vcenterreceiver panics on VM metric collection (nil pointer dereference in buildVMMetrics)

The otelcol-contrib process crashes during VM metric collection from a vCenter instance with a nil pointer dereference in buildVMMetrics. It appears to be caused by an unexpected nil in the VM object returned by the vSphere API (likely vm.Config.Hardware or vm.Summary fields).

Logs:
panic: runtime error: invalid memory address or nil pointer dereference
goroutine 147 [running]:
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End(...)
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver.(*vcenterMetricScraper).buildVMMetrics
        processors.go:310
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver.(*vcenterMetricScraper).processVMs
        processors.go:228
...



Config 
receivers:
  vcenter:
    endpoint: https://<vcenter>
    username: ...
    password: ...
    collection_interval: 60s
    insecure: true

processors:
  batch:

exporters:
  prometheus:
    endpoint: "0.0.0.0:9464"

service:
  pipelines:
    metrics/vcenter:
      receivers: [vcenter]
      processors: [batch]
      exporters: [prometheus]



Request
Please guard against nil dereferences in buildVMMetrics to avoid crashing the entire collector. A panic on one VM should not take down the pipeline.

--- a/.chloggen/main.yaml
+++ b/.chloggen/main.yaml
@@ -1,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: vcenterreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fixes a potential NPE during VM collection
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [41018]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []

--- a/receiver/vcenterreceiver/processors.go
+++ b/receiver/vcenterreceiver/processors.go
@@ -307,7 +307,7 @@ func (v *vcenterMetricScraper) buildVMMetrics(
 	}
 
 	groupInfo = &vmGroupInfo{poweredOff: 0, poweredOn: 0, suspended: 0, templates: 0}
-	if vm.Config.Template {
+	if vm.Config != nil && vm.Config.Template {
 		groupInfo.templates++
 	} else {
 		switch vm.Runtime.PowerState {

